paddle/cinn/auto_schedule/search_space/auto_gen_rule/reduction_factoring.cc

```diff
@@ -161,22 +161,42 @@ void ReductionFactoring::Apply(const std::string& block_name,
   // 5. Split the reduction loop into 2 part
   VLOG(6) << "before Split: " << ir_schedule->GetModule().GetExprs()[0];
   int factor = 1;
+  int max_factor = 1024;
   int extent = ir::GetLoopExtent(fused_reduce_loop);
-  for (int i = ceil(sqrt(extent)); i >= 1; --i) {
+  for (int i = max_factor; i >= 1; --i) {
```

Comment:
Here we change the speed up logic from `ceil(sqrt(extent))` to `max_factor`, why?

Should it be `std::min(max_factor, ceil(sqrt(extent)))` ?

Reply:
Here, the reduce loop will be divided into two parts: (extent / factor) and (factor), with the first part performing a serial reducction and the second part performing a parallel tree reduction. The current simple strategy is to hope that the second part is as large as possible, which can improve parallelism and reduce the probability of accuracy errors caused by adding large and small numbers.
Of course, currently these strategies are relatively naive, and in the future, we need more complex strategies to improve performance.

Reply:
Good